pandaharvester/harvesterbody/worker_adjuster.py

```diff
@@ -256,6 +274,26 @@ def define_num_workers(self, static_num_workers, site_name):
 
                             # Reset the maxQueueWorkers according to particular
                             if n_new_workers_def is not None:  # don't surpass limits given centrally
+
+                                # in slave mode
+                                if queue_config.mapType == "NoJob":
+                                    try:
+                                        if n_new_workers_def > 0:
+                                            activate_worker_factor = self.get_activate_worker_factor(queue_name, job_type, resource_type, queue_dict, queue_config)
+                                            tmp_log.debug(f"n_new_workers_def: {n_new_workers_def}, activate_worker_factor: {activate_worker_factor}")
+                                            n_new_workers_def = math.ceil(n_new_workers_def * activate_worker_factor)
+                                            tmp_log.debug(f"after applying activate_worker_factor, n_new_workers_def: {n_new_workers_def}")
+                                            if job_stats is not None:
+                                                # if job_stats is None, maybe job statistics is not fetched from panda, do nothing
+                                                queue_activated = job_stats.get(queue_name, {}).get("activated", 0)
+                                                n_queue_ready_running = n_queue + n_ready + n_running
+                                                tmp_log.debug(f"available activated panda jobs {queue_activated}, n_queue + n_ready + n_running: {n_queue_ready_running}")
+                                                if queue_activated < 1 and n_queue_ready_running > 0:
+                                                    tmp_log.info("No activated jobs and there are jobs in (queue, ready, running) status, set n_new_workers_def to 0")
+                                                    n_new_workers_def = 0
```

Comment:
@wguanicedew In slave mode n_new_workers_def is derived from PanDA's pilot streaming result, where activated jobs are already taken into account.
Also, job_stats on harvester is cached and may be not very up-to-date compared with those on PanDA. We have planned to use it for pure PULL (not slave mode) only since there is no other simpler mechanism to respect activated jobs.
Why the need to check activated jobs from job_stats in slave mode and even overwrite n_new_workers_def based on it?